src/components/button/button.ts

```diff
@@ -32,49 +43,55 @@ export class MdButton {
     setTimeout(() => { this.isMouseDown = false; }, 100);
   }
 
-  onFocus() {
+  setKeyboardFocus() {
     this.isKeyboardFocused = !this.isMouseDown;
   }
 
-  onBlur() {
+  removeKeyboardFocus() {
     this.isKeyboardFocused = false;
   }
 }
 
-
 @Component({
-  selector: 'a[md-button], a[md-raised-button], a[md-fab]',
-  inputs: ['disabled'],
+  selector: 'a[md-button], a[md-raised-button], a[md-fab], a[md-mini-fab]',
+  inputs: ['color'],
   host: {
-    '(click)': 'onClick($event)',
-    '(mousedown)': 'onMousedown()',
-    '(focus)': 'onFocus()',
-    '(blur)': 'onBlur()',
-    '[tabIndex]': 'tabIndex',
     '[class.md-button-focus]': 'isKeyboardFocused',
-    '[attr.aria-disabled]': 'isAriaDisabled',
+    '[class]': 'setClassList()',
+    '(mousedown)': 'setMousedown()',
+    '(focus)': 'setKeyboardFocus()',
+    '(blur)': 'removeKeyboardFocus()'
   },
   templateUrl: './components/button/button.html',
   styleUrls: ['./components/button/button.css'],
   encapsulation: ViewEncapsulation.None
 })
 export class MdAnchor extends MdButton implements OnChanges {
-  tabIndex: number;
-  disabled_: boolean;
+  disabled_: boolean = null;
+
+  @HostBinding('tabIndex') tabIndex: number;
 
-  get disabled(): boolean {
-    return this.disabled_;
+  @HostBinding('attr.aria-disabled')
+  /** Gets the aria-disabled value for the component, which must be a string for Dart. */
+  get isAriaDisabled(): string {
+    return this.disabled_ ? 'true' : 'false';
   }
 
-  set disabled(value) {
+  @HostBinding('attr.disabled')
+  @Input('disabled')
+  get disabled() { return this.disabled_; }
+
+  set disabled(value: boolean) {
     // The presence of *any* disabled value makes the component disabled, *except* for false.
-    this.disabled_ = value != null && this.disabled !== false;
+    this.disabled_ = (value !== null && value !== false) ? true : null;
```

Comment:
You actually want `value != null` here, which will also capture `undefined`. Why does disabled get set to null in some cases?